### PR TITLE
CLN:Unused Variables

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

Used PyCharm inspections to check:

- Unused local variables
- Unreachable code
- `set([..])` to `{...}`

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]
there are some very aggressive changes here

File: pandas/_libs/internals.pyx [hunk 1]
```diff
@@ -390,7 +390,7 @@ def get_blkno_indexers(int64_t[:] blknos, bint group=True):
     start = 0
     cur_blkno = blknos[start]
 
-    if group == False:
+    if not group:
```
Comment (jbrockmendel) [inline pandas/_libs/internals.pyx]: Group is false

File: pandas/compat/pickle_compat.py [hunk 2]
```diff
@@ -15,9 +15,6 @@ def load_reduce(self):
     args = stack.pop()
     func = stack[-1]
 
-    if len(args) and type(args[0]) is type:
-        n = args[0].__name__  # noqa
```
Comment (jbrockmendel) [inline pandas/compat/pickle_compat.py]: Do we know what this is for? “Noqa” suggests author knewflake would complain

File: pandas/core/generic.py [hunk 3]
```diff
@@ -1022,9 +1022,6 @@ def rename(self, *args, **kwargs):
         copy = kwargs.pop('copy', True)
         inplace = kwargs.pop('inplace', False)
         level = kwargs.pop('level', None)
-        axis = kwargs.pop('axis', None)
```
Comment (jbrockmendel) [inline pandas/core/generic.py]: I think part of the point of this is to empty out the kwargs dict, no?

**Comment by jbrockmendel** [comment]
I thought flake8 checked for unused variables, unreachable code.  Are we using an old version or something?

The set comprehensions are a good fix.  Might be worth adding that flake8 plugin while you’re at it.

**Comment by mroeschke** [comment]
I'm not sure if PyCharm is using flake8 under the hood or they have their own introspection method, but it was catching some things that flake8 wasn't.

Nonetheless, I'll scale this back and break it into smaller PR's

**Pull request closed without merging**